### PR TITLE
Remove all noImplicitAny warnings

### PR DESCRIPTION
**Pull request by mckoss**
The goal of this PR is to more strongly type all variables and arguments by setting the noImplicitAny Typescript configuration.

### CODE REVIEW

File: docs/todo.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Known issues and tasks.
 
+- Offline storage simulator (fake)
+  - Enable online an offline tests for rules simulator
```
Comment (mikelehen) [inline docs/todo.md]: s/an/and/

File: typings/main/ambient/node/node.d.ts [hunk 2]
```diff
@@ -0,0 +1,2196 @@
+// Compiled using typings@0.6.8
```
Comment (mikelehen) [inline typings/main/ambient/node/node.d.ts]: Do you intentionally have 2 node.d.ts files?  (one under typings/browser and one under typings/main)

**Comment by mikelehen** [comment]
Changes are too big to review in detail, but I skimmed through and overall looks reasonable to me!

File: typings/main/ambient/node/node.d.ts (hunk 2 above)
Comment (mckoss) [inline typings/main/ambient/node/node.d.ts]: I think typings does that.  I'm not clear on when one is used over the other.

**Pull request merged**